src/upgrade/nc_upgrade_scripts/1.0.0/config_dir_restructure.js

```diff
@@ -202,38 +200,36 @@ async function create_account_name_index_if_missing(config_fs, account_upgrade_p
  * @param {*} dbg 
  * @returns {Promise<Void>}
  */
-async function create_account_access_keys_index_if_missing(config_fs, account_upgrade_params, tmp_access_keys_path, dbg) {
+async function create_account_access_keys_index_if_missing(config_fs, account_upgrade_params, dbg) {
     const { fs_context } = config_fs;
     const { access_keys, _id, identity_path } = account_upgrade_params;
+    const backup_access_keys_path = path.join(config_fs.config_root, '.backup_access_keys_dir/');
 
+    await config_fs.create_dir_if_missing(backup_access_keys_path);
     if (access_keys) {
         for (const { access_key } of access_keys) {
             const access_key_path = config_fs.get_account_or_user_path_by_access_key(access_key);
-            const tmp_access_key_path = path.join(tmp_access_keys_path, native_fs_utils.get_config_files_tmpdir());
+            const tmp_access_key_path = path.join(backup_access_keys_path, access_key);
             const account_config_relative_path = config_fs.get_account_relative_path_by_id(_id);
 
-            const access_key_already_linked = await config_fs._is_symlink_pointing_to_identity(access_key_path, identity_path);
-            if (access_key_already_linked) continue;
+            try {
+                const access_key_already_linked = await config_fs._is_symlink_pointing_to_identity(access_key_path, identity_path);
+                if (access_key_already_linked) continue;
+            } catch (err) {
+                if (err.code === 'ENOENT') dbg.warn(`account access key is missing, re-creating ${access_key}`);
+            }
 
             try {
                 await nb_native().fs.symlink(fs_context, account_config_relative_path, tmp_access_key_path);
             } catch (err) {
                 if (err.code !== 'EEXIST') throw err;
                 dbg.warn(`account access key backup was already linked on a previous run of the upgrade script, continue ${access_keys}, ${tmp_access_key_path}`);
             }
-            let src_file;
             try {
-                if (native_fs_utils._is_gpfs(fs_context)) {
-                    src_file = await nb_native().fs.open(fs_context, tmp_access_key_path, 'r', native_fs_utils.get_umasked_mode(config.BASE_MODE_CONFIG_FILE));
-                    await src_file.linkfileat(fs_context, access_key_path);
-                } else {
-                    await nb_native().fs.rename(fs_context, tmp_access_key_path, access_key_path);
-                }
+                await nb_native().fs.rename(fs_context, tmp_access_key_path, access_key_path);
```

Comment:
Maybe add a comment about the usage of rename? (I wrote about it in the Pr description)

Reply:
I added it in the code -  this will replace atomically the old symlink with the new one